preferred-operator/interface.js

```diff
@@ -0,0 +1,366 @@
+  selectedOperator = menuButton.parentNode;
+};
+
+/**
+ * Closes the priority menu
+ */
+const closeMenu = () => {
+  const menu = document.getElementById('priority-menu');
+  const menuAction = document.querySelector('.priority-menu-action.active');
+
+  menuAction.classList.remove('active');
+  menu.classList.remove('active');
+};
+
+/**
+ * Set the priority of the operator that was clicked to any of the items that was clicked inside the menu
+ * @param { number } index - The index of the item that was clicked inside the menu
+ */
+const setPriority = index => {
+  switch (index) {
+    case 0:
+      // Unset the priority
+      priorities.delete(selectedOperator.dataset.id);
+      break;
+    case 1:
+      // Set the priority to high
+      priorities.set(selectedOperator.dataset.id, {
+        priority: 'high',
+        color: 'priority',
+        icon: 'light-high-priority',
+        label: 'High priority',
+      });
+      break;
+    case 2:
+      // Set the priority to medium
+      priorities.set(selectedOperator.dataset.id, {
+        priority: 'medium',
+        color: 'priority',
+        icon: 'light-medium-priority',
+        label: 'Medium priority',
+      });
+      break;
+    case 3:
+      // Set the priority to low
+      priorities.set(selectedOperator.dataset.id, {
+        priority: 'low',
+        color: 'priority',
+        icon: 'light-low-priority',
+        label: 'Low priority',
+      });
+      break;
+    default:
+      // Set the priority to excluded
+      priorities.set(selectedOperator.dataset.id, {
+        priority: 'excluded',
+        color: 'excluded',
+        icon: 'light-exclude-priority',
+        label: 'Excluded priority',
+      });
+      break;
+  }
+
+  // When we changed the priority we let the UI reflect it by calling this function
+  setPriorityStyling();
+};
+
+/**
+ * Sets the correct styling based on the priority that was selected inside the priority menu.
+ */
+const setPriorityStyling = () => {
+  const priority = priorities.get(selectedOperator.dataset.id);
+
+  /* eslint-disable */
+  selectedOperator.querySelector('.priority-list-icon').className = `priority-list-icon ${
+    priority ? priority.color : ''
+  }`;
+  selectedOperator.querySelector('.priority-list-icon img').src = `images/${
+    priority ? priority.icon : 'light-no-priority'
+  }.svg`;
+  selectedOperator.querySelector('.priority-label').innerHTML = `${priority ? priority.label : 'No priority'}`;
+  /* eslint-enable */
+};
+
+/**
+ * This recalculates the route based upon the new operator selection.
+ * It extracts the priorities that were set in the list and formats them to how our API is expecting the data
+ * If there are no priorities it just calculates a regular route.
+ */
+const recalculateRoute = () => {
+  loadingToast.style.transform = `translateY(0)`;
+
+  if (priorities.size > 0) {
+    const level1 = [];
+    const level2 = [];
+    const level3 = [];
+    const exclude = [];
```

Comment:
Perhaps explain that using a lower priority for the lowest priority (fe. 10), would make it less likely that the operator is used together with a link to the full documentation.